lib/action_view/component/base.rb

```diff
@@ -2,18 +2,29 @@
 
 # Monkey patch ActionView::Base#render to support ActionView::Component
 #
-# Upstreamed in https://github.com/rails/rails/pull/36388
-# Necessary for Rails versions < 6.1.0.alpha
+# A version of this monkey patch was upstreamed in https://github.com/rails/rails/pull/36388
+# We'll need to upstream an updated version of this eventually.
 class ActionView::Base
   module RenderMonkeyPatch
-    def render(component, _ = nil, &block)
-      return super unless component.respond_to?(:render_in)
-
-      component.render_in(self, &block)
+    def render(options = {}, args = {}, &block)
+      if options.respond_to?(:render_in)
+        ActiveSupport::Deprecation.warn(
+          "passing component instances to `render` will be deprecated in actionview-component 2.0. Use `render MyComponent, foo: :bar` instead."
```

Comment:
I am finding the wording
> will be deprecated in actionview-component 2.0

here (and in the `CHANGELOG` and `README`) a bit confusing.

Since this is already logging a deprecation warning, it sounds like this is deprecated _now_ so is the plan that this syntax will be _removed_ in v2.0?

Reply:
Oh, looks like you just noticed this as well and pushed 54bfb4b704ec3c69b2e882e528ef7024df0f0787

However, the confusing wording is still here:
https://github.com/github/actionview-component/blob/54bfb4b704ec3c69b2e882e528ef7024df0f0787/CHANGELOG.md#L5
and here:
https://github.com/github/actionview-component/blob/54bfb4b704ec3c69b2e882e528ef7024df0f0787/README.md#L148

Reply:
Resolved by 885bc97b218c1899d5adf16c68e32bd54aa10eed